src/workspaces/common/WorkspaceSelector.tsx

```diff
@@ -0,0 +1,61 @@
+import _ from 'lodash/fp';
+import React from 'react';
+import { CloudProviderIcon } from 'src/components/CloudProviderIcon';
+import { SelectProps, VirtualizedSelect } from 'src/components/common';
+import { getCloudProviderFromWorkspace, WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+interface WorkspaceSelectorProps
+  extends Omit<SelectProps<string, false, { value: string; label: string }>, 'value' | 'options'> {
+  workspaces: Workspace[];
+  value: string | undefined;
+  onChange: (string) => void;
+  id?: string | undefined;
+  'aria-label'?: string | undefined;
+}
+
+interface WorkspaceSelectorOption {
+  'aria-label': string;
+  value: string;
```

Comment:
Also, `formatOptionLabel` doesn't use value.

I would be tempted to just make this a render function within `WorkspaceSelector` with 2 typed input arguments-- `label` and `workspace`. Or even inline it where it's used.

Reply:
I would counter that it's better to have stuff broken out like this so we don't make the consuming component larger.  The one thing I would suggest is to capitalize to FormatOptionLabel and have it have the signature of a full funtional component.